common/cpp/tests/Makefile

```diff
@@ -36,10 +37,21 @@ build/%.o: %.cpp
 build/%.o: ../%.cpp
 	g++ -o $@ $(CPPFLAGS) -c $^
 
-build/%.o: ../crypto/%.cpp
+build/%.o: ../packages/base64/%.cpp
 	g++ -o $@ $(CPPFLAGS) -c $^
 
-build/%.o: ../packages/base64/%.cpp
+# Library-specific source has precedence over generic source in ../crypto/
+ifdef CRYPTOLIB_OPENSSL
+build/%.o: ../crypto/openssl/%.cpp
+	g++ -o $@ $(CPPFLAGS) -c $^
+endif
+
+ifdef CRYPTOLIB_MBEDTLS
```

Comment:
Is it possible to define both CRYPTOLIB_OPENSSL and CRYPTOLIB_MBEDTLS together or either one of them ?

Reply:
Yes. I will add a check that both are not defined at the same time.

Reply:
If we define both do we get any errors like multiple symbols defined for functions "RandomBitString" ?

Reply:
@manojgop I added a check for not defining both in `crypto_shared.h`.
There are no multiple symbols because only one or the other file is compiled with `test/Makefile` (not both).